packages/dashboard/src/dashboard-actions/annotations.ts

```diff
@@ -0,0 +1,89 @@
+import { XAnnotation } from '@synchro-charts/core';
+import { DashboardConfiguration } from '../types';
+
+export const addXAnnotation = ({
+  dashboardConfiguration,
+  widgetId,
+  annotation,
+}: {
+  dashboardConfiguration: DashboardConfiguration;
+  widgetId: string;
+  annotation: XAnnotation;
+}) => {
+  return dashboardConfiguration.map((widget) => {
+    if (widget.id == widgetId) {
+      let currAnnotations = widget.annotations;
+      if (currAnnotations && currAnnotations.x) {
+        currAnnotations.x.push(annotation);
+      } else if (!currAnnotations) {
+        currAnnotations = {
+          x: [annotation],
+        };
+      } else {
+        currAnnotations = {
+          ...currAnnotations,
+          x: [annotation],
+        };
+      }
+      return {
+        ...widget,
+        annotations: currAnnotations,
+      };
+    }
+    return widget;
+  });
+};
+
+export const deleteXAnnotation = ({
+  dashboardConfiguration,
+  widgetId,
+  annotationIdToDelete,
+}: {
+  dashboardConfiguration: DashboardConfiguration;
+  widgetId: string;
+  annotationIdToDelete: string;
+}) => {
+  return dashboardConfiguration.map((widget) => {
+    if (widget.id == widgetId) {
```

Comment:
should use `===`. It would read better if you flip the check and return earlier so you don't need to wrap the delete code in a block. for eg.

```
if (widget.id !== widgetId) return widget;
// do all the other manipulation you need to do to the widget
return {
...widget,
annotations
}
```

Reply:
Refactored to fit this scheme